### PR TITLE
chore: remove trait bound Hash from AccountId

### PR DESCRIPTION
**Pull request by neutrinoks**

### CODE REVIEW

File: src/balance.rs [hunk 1]
```diff
@@ -105,16 +107,19 @@ where
...
         let cheques = self.cheques.borrow();
         let mut purse = NegativeImbalanceOf::<T>::zero();
-        let mut depts = Vec::<(&AccountIdOf<T>, BalanceOf<T>)>::new();
+        let mut depts = Vec::<(AccountIdOf<T>, BalanceOf<T>)>::new();
 
         // Calculate balance differences and withdraw negative ones from user's accounts.
         for (account, balance) in cheques.iter() {
             let true_balance = self.initial_state.get(account).unwrap_or(&zero);
+            // let mut ref_acc: &[u8] = account;
+            // let account_id = T::AccountId::decode(&mut ref_acc)
```
Comment (Rqnsom) [inline src/balance.rs:116]: Do we want to merge these? :male-detective:

File: src/balance.rs [hunk 2]
```diff
@@ -155,7 +160,7 @@ where
 
     /// Does a state checking on initial state of cheques with current state.
     fn cmp_with_initial_state(&self) -> DispatchResult {
-        let cheques: Ref<HashMap<AccountIdOf<T>, BalanceOf<T>>> = self.cheques.borrow();
+        let cheques: Ref<HashMap<Vec<u8>, BalanceOf<T>>> = self.cheques.borrow();
```
Comment (Rqnsom) [inline src/balance.rs:163]: We could use an alias here to give the better context
`type EncodedAccount = Vec<u8>;`

```suggestion
        let cheques: Ref<HashMap<EncodedAccount>, BalanceOf<T>>> = self.cheques.borrow();
```

File: src/balance.rs [hunk 3]
```diff
@@ -233,3 +236,9 @@ where
         Ok(amount)
     }
 }
+
+fn bytevec2accountid<T: SysConfig>(vec: &Vec<u8>) -> Result<AccountIdOf<T>, DispatchError> {
```
Comment (Rqnsom) [inline src/balance.rs:240]: looks too tight without any underscore
```suggestion
fn vec_to_account_id<T: SysConfig>(vec: &Vec<u8>) -> Result<AccountIdOf<T>, DispatchError> {
```
or just try to implement TryFrom here if you have time

**Review by Rqnsom — COMMENTED** [review]
nice! 👍

**Review by Rqnsom — DISMISSED** [review]

File: src/balance.rs (hunk 2 above)
Comment (neutrinoks) [reply src/balance.rs:163]: Love it :+1:
Done

File: src/balance.rs (hunk 3 above)
Comment (neutrinoks) [reply src/balance.rs]: Done

File: src/balance.rs (hunk 1 above)
Comment (neutrinoks) [reply src/balance.rs]: Of course not! :laughing:
Done :+1:

**Review by Rqnsom — APPROVED** [review]
let's go! 🚀

**Pull request merged**